sui/operators.js

```diff
@@ -0,0 +1,297 @@
+const { Command, Option } = require('commander');
+const { Transaction } = require('@mysten/sui/transactions');
+const { bcs } = require('@mysten/sui/bcs');
+const { printInfo, loadConfig } = require('../common/utils');
+const { operatorsStruct } = require('./types-utils');
+const { addBaseOptions, addOptionsToCommands, parseSuiUnitAmount } = require('./cli-utils');
+const { getWallet, printWalletInfo, broadcast } = require('./sign-utils');
+const { getBcsBytesByObjectId, findOwnedObjectId } = require('./utils');
+
+async function getGasCollectorCapId(client, gasServiceConfig, contractConfig) {
+    const operatorId = contractConfig.objects.Operators;
+
+    // Get and parse operator data
+    const operatorBytes = await getBcsBytesByObjectId(client, operatorId);
+    const parsedOperator = operatorsStruct.parse(operatorBytes);
+
+    // Get the capabilities bag ID
+    const bagId = parsedOperator.caps.id;
+
+    // Find the GasCollectorCap bag ID
+    const bagResult = await client.getDynamicFields({
+        parentId: bagId,
+        name: 'caps',
+    });
+    const gasCollectorBagId = bagResult.data.find(
+        (cap) => cap.objectType === `${gasServiceConfig.address}::gas_service::GasCollectorCap`,
+    )?.objectId;
+
+    if (!gasCollectorBagId) {
+        throw new Error('GasCollectorCap not found in the operator capabilities bag');
+    }
+
+    // Get the actual cap ID from the bag ID
+    const gasCollectorCapObject = await client.getObject({
+        id: gasCollectorBagId,
+        options: {
+            showContent: true,
+        },
+    });
+
+    // Extract and return the gas collector cap ID
+    const gasCollectorCapId = gasCollectorCapObject.data.content.fields.value.fields.id.id;
+    return gasCollectorCapId;
+}
+
+async function collectGas(keypair, client, gasServiceConfig, contractConfig, args, options) {
+    const [amount] = args;
+    const receiver = options.receiver || keypair.toSuiAddress();
+
+    const operatorId = contractConfig.objects.Operators;
+    const gasCollectorCapId = await getGasCollectorCapId(client, gasServiceConfig, contractConfig);
```

Comment:
Since it can be read from the gas service config, helper use is not needed?

Reply:
yes, we can read it from the config. do you think we can remove the `getGasCollectorCapId ` function or we can convert it to a util function to read the bag content id?